docs(core): restructure & update intro page (#18441)
nrwl/nx: docs/shared/getting-started/intro.md

# Intro to Nx

Nx is a powerful open-source build system that provides tools and techniques for enhancing developer productivity, optimizing CI performance, and maintaining code quality.

## Core Features

> - **Run Tasks Efficiently**: Nx [runs tasks in parallel](/core-features/run-tasks) also considering dependencies between the tasks.

Reviewer: ```suggestion
- **Run Tasks Efficiently**: Nx [runs tasks in parallel](/core-features/run-tasks) and orders the tasks based on the dependencies between them.
```

- **Cache Locally & Remotely**: With [local](/core-features/cache-task-results) and [remote caching](/core-features/remote-cache), Nx prevents unnecessary re-runs of tasks, saving you valuable dev time.

> - **Automate Dependency Update**: if you leverage Nx plugins you gain additional features such as [code generation](/core-features/plugin-features/use-code-generators) and tools to [automatically upgrade](core-features/automate-updating-dependencies) your codebase and dependencies.

Reviewer: ```suggestion
- **Automate Dependency Updates**: if you leverage Nx plugins you gain additional features such as [code generation](/core-features/plugin-features/use-code-generators) and tools to [automatically upgrade](core-features/automate-updating-dependencies) your codebase and dependencies.
```

> - **Make it Your Own**: Nx is highly customizable and extensible. Fine-tune it by [create your own plugins](/extending-nx/intro/getting-started) and optionally even [share them with the community](/extending-nx/tutorials/publish-plugin#publish-your-nx-plugin).

Reviewer: ```suggestion
- **Make it Your Own**: Nx is highly customizable and extensible. Fine-tune it by [creating your own plugins](/extending-nx/intro/getting-started) and optionally [share them with the community](/extending-nx/tutorials/publish-plugin#publish-your-nx-plugin).
```

<!-- - **Monorepo and Single Projects**: Nx supports both, monorepos as well as single-project (standalone) workspaces. -->

> Find out more about [why you should use Nx](/getting-started/why-nx) or browse our [core features](/core-features) section.

Reviewer: ```suggestion
Find out more about [why you should use Nx](/getting-started/why-nx) or browse our [core features](/core-features).
```

> ## Try Yourself!

Reviewer: ```suggestion
## Try Nx Yourself!
```

Author: this works, but I was actually intentionally saying "Try Yourself" as a more direct call to action. But with "Nx" in the middle works too ofc

```shell
npx create-nx-workspace@latest
```

## Learn Nx

{% cards cols="4" %}

{% link-card title="Nx in 5 minutes!" type="video" url="/getting-started/tutorials/package-based-repo-tutorial" icon="/documentation/shared/images/icons/nx-color.svg" /%}

{% link-card title="Create a Package-based Monorepo" type="tutorial" url="/getting-started/tutorials/package-based-repo-tutorial" icon="/documentation/shared/images/icons/javascript-color.svg" /%}

{% link-card title="Create a Monorepo with Nx Plugins" type="tutorial" url="/getting-started/tutorials/integrated-repo-tutorial" icon="/documentation/shared/images/icons/javascript-color.svg" /%}

{% link-card title="On our YT Channel" type="video" url="https://www.youtube.com/@nxdevtools" icon="/documentation/shared/images/icons/youtube-color.svg" /%}

{% /cards %}

> ## Know Already? Pick Your Stack!

Reviewer: ```suggestion
## Pick Your Stack!
```

Reviewer: "Know Already?" has no clear object in the sentence.  If you're trying to ask if they know Nx already, that's already implied by the section above being "Learn Nx".  Another option is to number the headers: 1. Learn Nx.  2. Pick Your Stack!

I also don't want to have "Know Nx Already?" on the intro page, because nobody can self-diagnose if they know Nx.  They know some things about it, but how much are you asking them to know?

Author: This "Know already" was more meant in the direction of "Know already what you want", like which stack so go ahead and pick one. Something like that
